scripts/test.sh

```diff
@@ -14,7 +14,7 @@ run_test_suite () {
     LOCATION=$2
     rm -rf $LOCATION
     mkdir -p $LOCATION
-    find . \( -name '*.gcno' -or -name '*.gcda' \) -and -not -path '*coverage*' -exec cp --parents \{\} "$LOCATION" \;
+    find . \( -name '*.gcno' -or -name '*.gcda' \) -and -not -path '*coverage*' -exec rsync -R \{\} "$LOCATION" \;
```

Comment:
I think this particular change is not a good idea. We're currently using a non-portable extension of `cp` (which is what you ran into), but your replacement is also non-portable. I'd recommend we instead leverage POSIX-specified variants to accomplish this.

1. `cp`'s `--parents` option does the same thing as `mkdir`'s `-p` (which *is* POSIX-specified); so we could do this:
   ```suggestion
       find . \( -name '*.gcno' -or -name '*.gcda' \) -and -not -path '*coverage*' -exec mkdir -p $(dirname \{\}) && cp \{\} "$LOCATION/\{\}" \;
   ```

That looks terrible, so I'd love to not do that. 

2. Instead, we could use `install`'s `-D` which works much more closely to `cp --parents` and is also POSIX-specified:
   ```suggestion
       find . \( -name '*.gcno' -or -name '*.gcda' \) -and -not -path '*coverage*' -exec install -D -m644 \{\} "$LOCATION/\{\}" \;
   ```
   `-m644` specifies the files should have `rw-r--r--` permissions which is, I believe, what we would want for all `gcno`/`gcda` files

@metalicjames thoughts?

Reply:
What was wrong with the script previously? This code hasn't changed in quite a while.

Reply:
`--parents` is not available on MacOS

Reply:
In talking with @AlexRamRam about this a bit more offline; I think it probably makes sense for us to switch to `rsync` and add it to `./scripts/configure.sh` to-be-installed.